engine/src/main/java/io/zeebe/engine/nwe/container/MultiInstanceBodyProcessor.java

```diff
@@ -0,0 +1,337 @@
+    expressionBehavior = bpmnBehaviors.expressionBehavior();
+    incidentBehavior = bpmnBehaviors.incidentBehavior();
+  }
+
+  @Override
+  public Class<ExecutableMultiInstanceBody> getType() {
+    return ExecutableMultiInstanceBody.class;
+  }
+
+  @Override
+  public void onActivating(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {
+
+    // verify that the input collection variable is present and valid
+    final Optional<List<DirectBuffer>> results = readInputCollectionVariable(element, context);
+    if (results.isEmpty()) {
+      return;
+    }
+
+    eventSubscriptionBehavior
+        .subscribeToEvents(element, context)
+        .ifRightOrLeft(
+            ok -> stateTransitionBehavior.transitionToActivated(context),
+            failure -> incidentBehavior.createIncident(failure, context));
+  }
+
+  @Override
+  public void onActivated(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {
+
+    final var loopCharacteristics = element.getLoopCharacteristics();
+    final Optional<List<DirectBuffer>> inputCollection =
+        readInputCollectionVariable(element, context);
+    if (inputCollection.isEmpty()) {
+      return;
+    }
+
+    final var array = inputCollection.get();
+    loopCharacteristics
+        .getOutputCollection()
+        .ifPresent(variableName -> initializeOutputCollection(context, variableName, array.size()));
+
+    if (array.isEmpty()) {
+      // complete the multi-instance body immediately
+      stateTransitionBehavior.transitionToCompleting(context);
+      return;
+    }
+
+    if (loopCharacteristics.isSequential()) {
+      final var firstItem = array.get(0);
+      createInnerInstance(element, context, firstItem);
+
+    } else {
+      array.forEach(item -> createInnerInstance(element, context, item));
+    }
+  }
+
+  @Override
+  public void onCompleting(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {
+
+    eventSubscriptionBehavior.unsubscribeFromEvents(context);
+
+    element
+        .getLoopCharacteristics()
+        .getOutputCollection()
+        .ifPresent(variableName -> stateBehavior.propagateVariable(context, variableName));
+
+    stateTransitionBehavior.transitionToCompleted(context);
+  }
+
+  @Override
+  public void onCompleted(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {
+
+    stateTransitionBehavior.takeOutgoingSequenceFlows(element, context);
+
+    stateBehavior.consumeToken(context);
+    stateBehavior.removeInstance(context);
+  }
+
+  @Override
+  public void onTerminating(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {}
+
+  @Override
+  public void onTerminated(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {}
+
+  @Override
+  public void onEventOccurred(
+      final ExecutableMultiInstanceBody element, final BpmnElementContext context) {}
```

Comment:
What is happening here?

Reply:
As you can see nothing :laughing: 
It is not yet implemented -> #4469